lib/app.js

```diff
@@ -72,9 +72,17 @@ class Container extends Component {
 
   render () {
     const { Component, props, url } = this.props
+    const Layout = Component.pageLayout || null
+    const component = Layout ? (
+      <Layout>
```

Comment:
@tz5514 What about passing the `props` and `url` to `Layout`? That way, the Layout component could make dynamic rendering decisions if needed. Could also pass `Component` in case it subsequently wanted to pass different props to the page component. That way, I think you'd have full control over rendering.

Reply:
That's a good idea. The layout probably has its own props (from hoc or other), 
so I think we can pass the page props by independent "pageProps". like this:
```js
<Layout pageProps={...props} url={url}>
   <Component {...props} url={url} />
 </Layout>
```

Reply:
That's a good idea. The layout probably has its own props (from hoc or other), 
so I think we can pass the page props by independent "pageProps". like this:
```js
<Layout pageProps={...props} url={url}>
   <Component {...props} url={url} />
 </Layout>
```

Reply:
@tz5514 i'd just spread the props from the page. The layout can't have any other props than these considering that you expose the layout as a static property 👌 A HOC might expose props on the layout. But passing them off using `{...props}` seems like the logical way to do it. Since then we can just say: A layout component receives the same props as a top level page. Arunoda, rauchg and me are still discussing if we want getInitialProps on layout components btw. I'm thinking of exposing it as a community-made HOC, since you might want getInitialProps to run at the same time as the page component's getInitialProps, or run them in sync.

Reply:
@timneutkens Yes! I actually want to put getInitialProps on layout components too.
Because there are some data actions being shared for different pages, like: SSR token auth.
It will be so convenience if we put getInitialProps on layout components.
BTW, I think run page and layout's getInitialProps in sync is more intuitional.

So it will be "a layout's props is decided by layout's getInitialProps" rather than "a layout component receives the same props as a top level page" I think.

Reply:
Okay, so here's what I'd like to do. Implement getInitialProps for the layout component. Then, if some wants to run them at the same time they can use this HOC I've drafted and change the awaits to a Promise.all.

```js
const layoutGetInitialProps = (ComposedComponent) => {
  const ComponentLayoutComponent = Component.layout
  const Layout = ({layoutProps}) => <ComponentLayoutComponent {...layoutProps} />

  return class extends React.Component {
    static async getInitialProps(ctx) {
      let layoutProps = {}
      let props = {}
      if(typeof ComponentLayoutComponent.getInitialProps === 'function') {
        layoutProps = await ComponentLayoutComponent.getInitialProps(ctx)
      }

      if(typeof ComposedComponent.getInitialProps === 'function') {
        props = await ComposedComponent.getInitialProps(ctx)        
      }

      return {layoutProps, ...props}
    }

    static layout = Layout

    render() {
      const props = {...this.props}
      delete props.layoutProps
      return <ComposedComponent {...props} />
    }
  }
}
```

Reply:
Also, this HOC proves the static layout property is just as flexible as the render function.